31_HLSLPathTracer/app_resources/hlsl/render_common.hlsl

```diff
@@ -0,0 +1,23 @@
+#ifndef _NBL_HLSL_PATHTRACER_RENDER_COMMON_INCLUDED_
+#define _NBL_HLSL_PATHTRACER_RENDER_COMMON_INCLUDED_
+
+struct SPushConstants
+{
+    float32_t4x4 invMVP;
+    int sampleCount;
+    int depth;
+};
+
+[[vk::push_constant]] SPushConstants pc;
+
+[[vk::combinedImageSampler]][[vk::binding(0, 2)]] Texture2D<float3> envMap;      // unused
+[[vk::combinedImageSampler]][[vk::binding(0, 2)]] SamplerState envSampler;
+
+[[vk::binding(1, 2)]] Buffer<uint3> sampleSequence;
+
+[[vk::combinedImageSampler]][[vk::binding(2, 2)]] Texture2D<uint2> scramblebuf; // unused
```

Comment:
please use the scramble buffer, it will allow us to do stuff like key-exchanges to attempt to get Blue Noise (as per some Heitz papers)